### PR TITLE
Switch to alpine golang docker image, bump golang + node versions, migrate to circleci v2

### PR DESCRIPTION
**Pull request by StephenRadachy**
- Switched to alpine 3.6 and yarn.
- Bumped golang and node versions.
- Switched to circleci v2 since it's new and because the updated dockerfile needs multi-stage support, which started in v17.05.

### CODE REVIEW

**Comment by StephenRadachy** [comment]
Right now, the new image is pretty hefty at 564MB, which is due to the addition of node. However, once the web ui is stripped out per https://github.com/loadimpact/k6/issues/300, the image size will shrink significantly to about 297MB. For context, golang:1.9-alpine itself is 270MB.

**Comment by liclac** [comment]
Wait, you can have multiple FROM's in Dockerfiles now?!

Anyhow, in light of #300, could you straight up remove the Node stuff from the image while we're anyways at it? It's adding a disproportionate amount of build time for something nobody is actually using.

**Comment by StephenRadachy** [comment]
@liclac all set. When this gets merged in, https://github.com/loadimpact/k6/issues/300 will be 95% done. The last thing will be to remove any looming references within the docs.

I wiped out anything that was related to the web ui. I ended up pointing the `/` route to the existing mux ping handler.

On a semi-unrelated note - builds are down to under 4 minutes 🎉

File: api/server.go [hunk 1]
```diff
@@ -24,21 +24,18 @@ import (
 	"fmt"
 	"net/http"
 
-	"github.com/GeertJohan/go.rice"
```
Comment (liclac) [inline api/server.go]: If this is removed, then it needs to be dropped from the `rice` command in `build-dist.sh` too.

Comment (StephenRadachy) [reply api/server.go]: @liclac ready for liftoff!

cat button press

**Pull request merged**